File: base/src/main/java/com/zhangteng/base/widget/CommonDialog.java
package com.zhangteng.base.widget;

import android.app.Dialog;
import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.text.SpannableString;
import android.text.TextUtils;
import android.text.method.LinkMovementMethod;
import android.view.View;
import android.widget.TextView;

import com.zhangteng.base.R;
import com.zhangteng.base.base.BaseDialog;
import com.zhangteng.base.utils.AntiShakeUtils;

public class CommonDialog extends BaseDialog implements View.OnClickListener {
    private TextView contentTxt;
    private TextView titleTxt;
    private TextView submitTxt;
    private TextView cancelTxt;
    private View devider;
    private View titleDevider;

    private Context mContext;
    private String content;
    private String positiveName;
    private String negativeName;
    private String title;
    private OnCloseListener listener;

    public CommonDialog(@NonNull Context context) {
        super(context);
        this.mContext = context;
    }

    public CommonDialog(Context context, String content) {
        super(context);
        this.mContext = context;
        this.content = content;
        if (!TextUtils.isEmpty(content))
            contentTxt.setText(content);
    }

    public CommonDialog setPositiveButton(String name) {
        this.positiveName = name;
        if (!TextUtils.isEmpty(positiveName)) {
            submitTxt.setText(positiveName);
        }
        return this;
    }

    public CommonDialog setNegativeButton(String name) {
        this.negativeName = name;
        if (!TextUtils.isEmpty(negativeName)) {
            cancelTxt.setText(negativeName);
            cancelTxt.setVisibility(View.VISIBLE);
            devider.setVisibility(View.VISIBLE);
        }
        return this;
    }

    public CommonDialog setTitle(String title) {
        this.title = title;
        if (!TextUtils.isEmpty(title)) {
            titleTxt.setText(title);
            titleTxt.setVisibility(View.VISIBLE);
            titleDevider.setVisibility(View.VISIBLE);
        }
        return this;
    }

    public CommonDialog setContent(String content) {
        this.content = content;
        if (!TextUtils.isEmpty(content)) {
            contentTxt.setText(content);
        }
        return this;
    }

    public CommonDialog setContent(SpannableString content) {
        this.content = content.toString();
        if (!TextUtils.isEmpty(content)) {
            contentTxt.setHighlightColor(Color.TRANSPARENT);
            contentTxt.setMovementMethod(LinkMovementMethod.getInstance());
            contentTxt.setText(content);
        }
        return this;
    }

    public CommonDialog setDialogCancelable(boolean flag) {
        setCancelable(flag);
        return this;
    }

    public CommonDialog setDialogCanceledOnTouchOutside(boolean cancel) {
        setCanceledOnTouchOutside(cancel);
        return this;
    }

    @Override
    public int getSelfTitleView() {
        return 0;
    }

    @Override
    public int getSelfContentView() {
        return R.layout.dialog_common;
    }

    @Override
    public int getSelfButtonView() {
        return R.layout.dialog_common_button;
    }

    @Override
    public void initView(View view) {
        contentTxt = view.findViewById(R.id.content);
        titleTxt = view.findViewById(R.id.title);
        titleDevider = view.findViewById(R.id.titleDevider);
        devider = view.findViewById(R.id.devider);
        submitTxt = view.findViewById(R.id.submit);
        submitTxt.setOnClickListener(this);
        cancelTxt = view.findViewById(R.id.cancel);
        cancelTxt.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if (AntiShakeUtils.isInvalidClick(v))
            return;
        if (v.getId() == R.id.cancel) {
            if (listener != null) {
                listener.onClick(this, false);
            }
            this.dismiss();
        } else if (v.getId() == R.id.submit) {
            if (listener != null) {
                listener.onClick(this, true);
            }
            this.dismiss();
        }
    }

    public CommonDialog setListener(OnCloseListener listener) {
        this.listener = listener;
        return this;
    }

    public interface OnCloseListener {
        void onClick(Dialog dialog, boolean confirm);
    }
}

File: base/src/main/java/com/zhangteng/base/widget/CommonDialog.kt
package com.zhangteng.base.widget

import android.app.Dialog
import android.content.Context
import android.graphics.Color
import android.text.SpannableString
import android.text.TextUtils
import android.text.method.LinkMovementMethod
import android.view.View
import android.widget.TextView
import com.zhangteng.base.R
import com.zhangteng.base.base.BaseDialog
import com.zhangteng.base.utils.AntiShakeUtils

open class CommonDialog : BaseDialog, View.OnClickListener {
    private var contentTxt: TextView? = null
    private var titleTxt: TextView? = null
    private var submitTxt: TextView? = null
    private var cancelTxt: TextView? = null
    private var devider: View? = null
    private var titleDevider: View? = null
    private var mContext: Context?
    private var content: String? = null
    private var positiveName: String? = null
    private var negativeName: String? = null
    private var title: String? = null
    private var listener: OnCloseListener? = null

    constructor(context: Context) : super(context) {
        mContext = context
    }

    constructor(context: Context, content: String?) : super(context) {
        mContext = context
        this.content = content
        if (!TextUtils.isEmpty(content)) contentTxt?.text = content
    }

    open fun setPositiveButton(name: String?): CommonDialog {
        positiveName = name
        if (!TextUtils.isEmpty(positiveName)) {
            submitTxt?.text = positiveName
        }
        return this
    }

    open fun setNegativeButton(name: String?): CommonDialog {
        negativeName = name
        if (!TextUtils.isEmpty(negativeName)) {
            cancelTxt?.text = negativeName
            cancelTxt?.visibility = View.VISIBLE
            devider?.visibility = View.VISIBLE
        }
        return this
    }

    open fun setTitle(title: String?): CommonDialog {
        this.title = title
        if (!TextUtils.isEmpty(title)) {
            titleTxt?.text = title
            titleTxt?.visibility = View.VISIBLE
            titleDevider?.visibility = View.VISIBLE
        }
        return this
    }

    open fun setContent(content: String?): CommonDialog {
        this.content = content
        if (!TextUtils.isEmpty(content)) {
            contentTxt?.text = content
        }
        return this
    }

    open fun setContent(content: SpannableString?): CommonDialog {
        this.content = content.toString()
        if (!TextUtils.isEmpty(content)) {
            contentTxt?.highlightColor = Color.TRANSPARENT
            contentTxt?.movementMethod = LinkMovementMethod.getInstance()
            contentTxt?.text = content
        }
        return this
    }

    open fun setDialogCancelable(flag: Boolean): CommonDialog {
        setCancelable(flag)
        return this
    }

    open fun setDialogCanceledOnTouchOutside(cancel: Boolean): CommonDialog {
        setCanceledOnTouchOutside(cancel)
        return this
    }

    override fun getSelfTitleView(): Int {
        return 0
    }

    override fun getSelfContentView(): Int {
        return R.layout.dialog_common
    }

    override fun getSelfButtonView(): Int {
        return R.layout.dialog_common_button
    }

    override fun initView(view: View?) {
        view?.let {
            contentTxt = view.findViewById(R.id.content)
            titleTxt = view.findViewById(R.id.title)
            titleDevider = view.findViewById(R.id.titleDevider)
            devider = view.findViewById(R.id.devider)
            submitTxt = view.findViewById(R.id.submit)
            submitTxt?.setOnClickListener(this)
            cancelTxt = view.findViewById(R.id.cancel)
            cancelTxt?.setOnClickListener(this)
        }
    }

    override fun onClick(v: View?) {
        v ?: return
        if (AntiShakeUtils.isInvalidClick(v)) return
        if (v.id == R.id.cancel) {
            listener?.onClick(this, false)
            dismiss()
        } else if (v.id == R.id.submit) {
            listener?.onClick(this, true)
            dismiss()
        }
    }

    open fun setListener(listener: OnCloseListener): CommonDialog {
        this.listener = listener
        return this
    }

    interface OnCloseListener {
        open fun onClick(dialog: Dialog?, confirm: Boolean)
    }
}